examples/material_gallery/lib/demo/weathers_demo.dart

```diff
@@ -0,0 +1,495 @@
+      )
+    );
+  }
+}
+
+List<Color> _kBackgroundColorsTop = [
+  new Color(0xff5ebbd5),
+  new Color(0xff0b2734),
+  new Color(0xffcbced7)
+];
+
+List<Color> _kBackgroundColorsBottom = [
+  new Color(0xff4aaafb),
+  new Color(0xff4c5471),
+  new Color(0xffe0e3ec)
+];
+
+class WeatherWorld extends NodeWithSize {
+  WeatherWorld() : super(new Size(2048.0, 2048.0)) {
+    _background = new GradientNode(
+      this.size,
+      _kBackgroundColorsTop[0],
+      _kBackgroundColorsBottom[0]
+    );
+    addChild(_background);
+
+    _cloudsSharp = new CloudLayer(_images['packages/flutter_gallery_assets/clouds-0.png'], false, false, 20.0);
+    addChild(_cloudsSharp);
+
+    _cloudsDark = new CloudLayer(_images['packages/flutter_gallery_assets/clouds-1.png'], true, true, 40.0);
+    addChild(_cloudsDark);
+
+    _cloudsSoft = new CloudLayer(_images['packages/flutter_gallery_assets/clouds-1.png'], false, false, 60.0);
+    addChild(_cloudsSoft);
+
+    _sun = new Sun();
+    _sun.position = new Point(1024.0, 1024.0);
+    _sun.scale = 1.5;
+    addChild(_sun);
+
+    _rain = new Rain();
+    addChild(_rain);
+
+    _snow = new Snow();
+    addChild(_snow);
+  }
+
+  GradientNode _background;
+  CloudLayer _cloudsSharp;
+  CloudLayer _cloudsSoft;
+  CloudLayer _cloudsDark;
+  Sun _sun;
+  Rain _rain;
+  Snow _snow;
+
+  WeatherType _weatherType = WeatherType.sun;
+
+  WeatherType get weatherType => _weatherType;
+
+  set weatherType(WeatherType weatherType) {
+    _weatherType = weatherType;
+
+    // Fade the background
+    _background.actions.stopAll();
+
+    _background.actions.run(new ActionTween(
+      (a) => _background.colorTop = a,
+      _background.colorTop,
+      _kBackgroundColorsTop[weatherType.index],
+      1.0
+    ));
+
+    _background.actions.run(new ActionTween(
+      (a) => _background.colorBottom = a,
+      _background.colorBottom,
+      _kBackgroundColorsBottom[weatherType.index],
+      1.0
+    ));
+
+    _cloudsDark.active = weatherType != WeatherType.sun;
+    _sun.active = weatherType == WeatherType.sun;
+    _rain.active = weatherType == WeatherType.rain;
+    _snow.active = weatherType == WeatherType.snow;
+  }
+
+  void spriteBoxPerformedLayout() {
+    _sun.position = spriteBox.visibleArea.topLeft + new Offset(350.0, 180.0);
+  }
+}
+
+class GradientNode extends NodeWithSize {
+  GradientNode(Size size, this.colorTop, this.colorBottom) : super(size);
+
+  Color colorTop;
+  Color colorBottom;
+
+  void paint(Canvas canvas) {
+    applyTransformForPivot(canvas);
+
+    Paint gradientPaint = new Paint()..shader = new LinearGradient(
+      begin: Point.origin,
+      end: new Point(0.0, size.height),
+      colors: <Color>[colorTop, colorBottom],
+      stops: <double>[0.0, 1.0]
+    ).createShader();
+
+    canvas.drawRect(new Rect.fromLTWH(0.0, 0.0, size.width, size.height), gradientPaint);
+  }
+}
+
+class CloudLayer extends Node {
+  CloudLayer(ui.Image image, bool dark, bool rotated, double loopTime) {
```

Comment:
make these named arguments, so that you cal tell what the bools are. Never have positioned bool arguments.